### PR TITLE
Add data sources for Managed Rules for WAF and WAF Regional

### LINKED ISSUE #12407: New Data Source: WAF Subscribed rule group
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

New data sources to fetch the waf (and waf regional) subscribed rule groups. These rule groups can be subscribed in the market place. 

Two data sources may be required because the same rule doesn't have the same ID depending on when using it with a regional web acl or a global web acl. 
```bash
$ aws waf-regional list-subscribed-rule-groups
{
    "RuleGroups": [
        {
            "RuleGroupId": "490f6909-95a7-4890-8ef7-cd2d2bb75950",
            "Name": "Fortinet Managed Rules for AWS WAF - Complete OWASP Top 10",
            "MetricName": "FortinetCompleteOWASPTopTenRuleset"
        }
    ]
}

$ aws waf list-subscribed-rule-groups
{
    "RuleGroups": [
        {
            "RuleGroupId": "fecbf80a-be95-43f2-a891-972c722f32f6",
            "Name": "Fortinet Managed Rules for AWS WAF - Complete OWASP Top 10",
            "MetricName": "FortinetCompleteOWASPTopTenRuleset"
        }
    ]
}
```

### New or Affected Resource(s)

* aws_waf_subscribed_rule_group
* aws_wafregional_subscribed_rule_group

### Potential Terraform Configuration

```hcl
data "aws_waf_subscribed_rule_group" "example" {
  name = ""
}

data "aws_wafregional_subscribed_rule_group" "example" {
  name = ""
}
```

### References
* API Reference: https://docs.aws.amazon.com/waf/latest/APIReference/API_waf_ListSubscribedRuleGroups.html
* SDK Reference: https://docs.aws.amazon.com/sdk-for-go/api/service/waf/#WAF.ListSubscribedRuleGroups and https://docs.aws.amazon.com/sdk-for-go/api/service/wafregional/#WAFRegional.ListSubscribedRuleGroups

### PR DESCRIPTION
**Pull request by flosell**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
**New Data Source:** `aws_waf_subscribed_rule_group`
**New Data Source:** `aws_wafregional_subscribed_rule_group` 
```

Output from acceptance testing 
```
$ AWS_WAF_SUBSCRIBED_RULE_GROUP_NAME="F5 Bot Detection Signatures For AWS WAF" AWS_WAF_SUBSCRIBED_RULE_GROUP_METRIC_NAME=F5BotDetectionSignatures make testacc TEST=./aws TESTARGS='-run=SubscribedRuleGroup_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=SubscribedRuleGroup_Basic -timeout 120m
=== RUN   TestAccDataSourceAwsWafSubscribedRuleGroup_Basic
=== PAUSE TestAccDataSourceAwsWafSubscribedRuleGroup_Basic
=== RUN   TestAccDataSourceAwsWafRegionalSubscribedRuleGroup_Basic
=== PAUSE TestAccDataSourceAwsWafRegionalSubscribedRuleGroup_Basic
=== CONT  TestAccDataSourceAwsWafSubscribedRuleGroup_Basic
=== CONT  TestAccDataSourceAwsWafRegionalSubscribedRuleGroup_Basic
--- PASS: TestAccDataSourceAwsWafSubscribedRuleGroup_Basic (111.39s)
--- PASS: TestAccDataSourceAwsWafRegionalSubscribedRuleGroup_Basic (112.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	112.769s
```

Notes on acceptance testing: 
* requires manually subscribing to a WAF rule product on the [marketplace](https://console.aws.amazon.com/wafv2/home?region=global#/partnerrules) first and then setting `AWS_WAF_SUBSCRIBED_RULE_GROUP_NAME` and `AWS_WAF_SUBSCRIBED_RULE_GROUP_METRIC_NAME` correctly; use `aws waf list-subscribed-rule-groups` to find what the account is already subscribed to
* the tests expect name and metric name to be for the same rule group
* a subscription applies for both WAF and WAF Regional (but use different rule group ids), therefore the same environment variable is used to test both data sources

Closes #12407

### CODE REVIEW

**Comment by DrFaust92** [comment]
Closes #12407

**Comment by haidaraM** [comment]
Looks like this PR is ready to be merged. Any plan to merge it soon :-) ?

**Comment by costingalan** [comment]
Any updates here?

**Comment by JnMik** [comment]
Would that allow to grab "Managed rule group" not from the marketplace as well ?

File: website/aws.erb [hunk 1]
```diff
@@ -3161,6 +3161,9 @@
                                 <li>
                                     <a href="/docs/providers/aws/d/waf_rate_based_rule.html">aws_waf_rate_based_rule</a>
                                 </li>
+                                <li>
+                                    <a href="/docs/providers/aws/d/waf_subscribed_rule_group.html">aws_waf_subscribed_rule_group</a>
+                                </li>
```
Comment (DrFaust92) [inline website/aws.erb:3166]: missing the second data source here

Comment (flosell) [reply website/aws.erb:3166]: Fixed! Thanks for catching this!

**Comment by flosell** [comment]
Fixed small documentation bug (thanks @DrFaust92 for spotting that!), resolved merge conflicts with most recent master and updated to latest linting and standards. Should be ready to merge again

**Comment by teamterraform** [comment]
## Notification of Recent and Upcoming Changes to Contributions

Thank you for this contribution! There have been a few recent development changes that affect this pull request. We apologize for the inconvenience, especially if there have been long review delays up until now. Please note that this is automated message from an unmonitored account. See the [FAQ](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/FAQ.md) for additional information on the maintainer team and review prioritization.

If you are unable to complete these updates, please leave a comment for the community and maintainers so someone can potentially continue the work. The maintainers will encourage other contributors to use the existing contribution as the base for additional changes as appropriate. Otherwise, contributions that do not receive updated code or comments from the original contributor may be closed in the future so the maintainers can focus on active items.

For the most up to date information about Terraform AWS Provider development, see the [Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/CONTRIBUTING.md). Additional technical debt changes can be tracked with the [`technical-debt` label on issues](https://github.com/terraform-providers/terraform-provider-aws/issues?q=is%3Aopen+is%3Aissue+label%3Atechnical-debt).

As part of updating a pull request with these changes, the most current unit testing and linting will run. These may report issues that were not previously reported.

### Action Required: Terraform 0.12 Syntax

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/8950
Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14417

Version 3 and later of the Terraform AWS Provider, which all existing contributions would potentially be added, only supports Terraform 0.12 and later. Certain syntax elements of Terraform 0.11 and earlier show deprecation warnings during runs with Terraform 0.12. Documentation and test configurations, such as those including deprecated string interpolations (`some_attribute = "${aws_service_thing.example.id}"`) should be updated to the newer syntax (`some_attribute = aws_service_thing.example.id`). Contribution testing will automatically fail on older syntax in the near future. Please see the referenced issues for additional information.

### Action Required: Terraform Plugin SDK Version 2

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14551

The Terraform AWS Provider has been upgraded to the latest version of the Terraform Plugin SDK. Generally, most changes to contributions should only involve updating Go import paths in source code files. Please see the referenced issue for additional information.

### Action Required: Removal of website/aws.erb File

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14712

Any changes to the `website/aws.erb` file are no longer necessary and should be removed from this contribution to prevent merge issues in the near future when the file is removed from the repository. Please see the referenced issue for additional information.

### Upcoming Change of Git Branch Naming

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14292

Development environments will need their upstream Git branch updated from `master` to `main` in the near future. Please see the referenced issue for additional information and scheduling.

### Upcoming Change of GitHub Organization

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14715

This repository will be migrating from https://github.com/terraform-providers/terraform-provider-aws to https://github.com/hashicorp/terraform-provider-aws. No practitioner or developer action is anticipated and most GitHub functionality will automatically redirect to the new location. Go import paths including `terraform-providers` can remain for now. Please see the referenced issue for additional information and scheduling.

**Comment by breathingdust** [comment]
Hi all 👋 Just letting you know that this is issue is featured on this quarters [roadmap](https://github.com/hashicorp/terraform-provider-aws/blob/main/ROADMAP.md). If a PR exists to close the issue a maintainer will review and either make changes directly, or work with the original author to get the contribution merged. If you have written a PR to resolve the issue please ensure the "Allow edits from maintainers" box is checked. Thanks for your patience and we are looking forward to getting this merged soon!

**Comment by zhelding** [comment]
Acceptance tests passing on latest commit 6787fc16267ae83e6a2d7c0079eb606b4b141e5f:

```
❯ make testacc TESTS=TestAccWAFSubscribedRuleGroupDataSource PKG=waf                  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/waf/... -v -count 1 -parallel 20 -run='TestAccWAFSubscribedRuleGroupDataSource'  -timeout 180m
=== RUN   TestAccWAFSubscribedRuleGroupDataSource_basic
=== PAUSE TestAccWAFSubscribedRuleGroupDataSource_basic
=== CONT  TestAccWAFSubscribedRuleGroupDataSource_basic
--- PASS: TestAccWAFSubscribedRuleGroupDataSource_basic (29.30s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/waf        29.335s

❯ make testacc TESTS=TestAccWAFRegionalSubscribedRuleGroupDataSource PKG=wafregional
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/wafregional/... -v -count 1 -parallel 20 -run='TestAccWAFRegionalSubscribedRuleGroupDataSource'  -timeout 180m
=== RUN   TestAccWAFRegionalSubscribedRuleGroupDataSource_basic
=== PAUSE TestAccWAFRegionalSubscribedRuleGroupDataSource_basic
=== CONT  TestAccWAFRegionalSubscribedRuleGroupDataSource_basic
--- PASS: TestAccWAFRegionalSubscribedRuleGroupDataSource_basic (33.98s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/wafregional        34.087s
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccWAFSubscribedRuleGroupDataSource_' PKG=waf ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/waf/... -v -count 1 -parallel 2  -run=TestAccWAFSubscribedRuleGroupDataSource_ -timeout 180m
=== RUN   TestAccWAFSubscribedRuleGroupDataSource_basic
    subscribed_rule_group_test.go:16: Environment variable WAF_SUBSCRIBED_RULE_GROUP_NAME is not set
--- SKIP: TestAccWAFSubscribedRuleGroupDataSource_basic (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/waf	7.296s
% make testacc TESTARGS='-run=TestAccWAFRegionalSubscribedRuleGroupDataSource_' PKG=wafregional ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/wafregional/... -v -count 1 -parallel 2  -run=TestAccWAFRegionalSubscribedRuleGroupDataSource_ -timeout 180m
=== RUN   TestAccWAFRegionalSubscribedRuleGroupDataSource_basic
    subscribed_rule_group_test.go:16: Environment variable WAF_SUBSCRIBED_RULE_GROUP_NAME is not set
--- SKIP: TestAccWAFRegionalSubscribedRuleGroupDataSource_basic (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/wafregional	5.221s
```

**Comment by zhelding** [comment]
Thanks so much for your contribution @flosell! Finally getting this merged :tada:

**Pull request merged**